### PR TITLE
Use createJsGraphBuilderFactoryFromConfig function from FrameworkUtils

### PR DESCRIPTION
**Pull request by RushanNanayakkara**

### CODE REVIEW

File: components/org.wso2.carbon.identity.conditional.auth.functions.notification/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/notification/SendEmailFunctionImplTest.java [hunk 1]
```diff
@@ -58,6 +59,7 @@ protected void setUp() throws Exception {
 
         super.setUp();
 
+        CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME = true;
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.conditional.auth.functions.notification/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/notification/SendEmailFunctionImplTest.java:62]: what is the usecase behind this?

File: components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml [hunk 2]
```diff
@@ -104,6 +104,19 @@
             <artifactId>nashorn-core</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.orbit.com.nimbusds</groupId>
+            <artifactId>nimbus-jose-jwt</artifactId>
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml:109]: why are we adding these new dependencies/

File: components/org.wso2.carbon.identity.conditional.auth.functions.notification/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/notification/SendEmailFunctionImplTest.java (hunk 1 above)
Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.conditional.auth.functions.notification/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/notification/SendEmailFunctionImplTest.java:62]: This is used in the pre test set ups. However there is no default value provided for this in CarbonConstants. Therefore this gives a NullPointerException when tests are run. Therefore the value was initialised here before tests run.

File: components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml (hunk 2 above)
Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml:109]: There were build issues due to this dependency not being present. There are multiple usages of this in the codebase however the dependency was missing. Therefore there were compilation issues.

File: components/org.wso2.carbon.identity.conditional.auth.functions.analytics/pom.xml [hunk 3]
```diff
@@ -109,6 +109,12 @@
             <groupId>org.wso2.orbit.org.apache.httpcomponents</groupId>
             <artifactId>httpasyncclient</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.organization.management.core</groupId>
+            <artifactId>org.wso2.carbon.identity.organization.management.service</artifactId>
+            <version>${identity.organization.management.core.version}</version>
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.conditional.auth.functions.analytics/pom.xml:115]: Why we need to define the version in the component pom file?

Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.conditional.auth.functions.analytics/pom.xml]: removed the versions from component pom files.

**Review by nilasini — APPROVED** [review]

**Comment by nilasini** [comment]
We have to trigger the PR builder before merging the PR
@RushanNanayakkara

**Comment by RushanNanayakkara** [comment]
> We have to trigger the PR builder before merging the PR @RushanNanayakkara

My bad. Will trigger a pr build for the version bump to test.

**Pull request merged**